### PR TITLE
Generate Client Code

### PR DESCRIPTION
**Pull request by thiunuwan**
## Purpose
Generated the client code for the Mistral API connector. 

Issue Title: Introduce a Ballerina connector for Mistral REST API
Issue link: https://github.com/ballerina-platform/ballerina-library/issues/7046

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

**Comment by ayeshLK** [comment]
@thiunuwan lets use following command to generate the connector. With this command the license header will be automatically added to the generated files.

```sh
bal openapi -i docs/spec/openapi.yml --mode client --license docs/license.txt -o ballerina
```

File: ballerina/tests/tests.bal [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+// Initialize the Mistral client with authentication and service URL
+ConnectionConfig mistralConfig = {auth: {token: mistralApiKey}};
+final Client mistral = check new Client(mistralConfig, mistralServiceUrl);
+
+
+// Test to get models in Mistral
+@test:Config {
+    groups: ["live_tests", "mock_tests"]
+}
+isolated function testGetModels() returns error? {
```
Comment (ayeshLK) [inline ballerina/tests/tests.bal:34]: Shall we add few more test cases ?

File: docs/spec/sanitations.md [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
 This document records the sanitation done on top of the official OpenAPI specification from Mistral AI. 
-The OpenAPI specification is obtained from (TODO: Add source link).
+The OpenAPI specification is obtained from [Mistral AI OpenAPI specification](https://docs.mistral.ai/redocusaurus/plugin-redoc-0.yaml).
 These changes are done in order to improve the overall usability, and as workarounds for some known language limitations.
 
 [//]: # (TODO: Add sanitation details)
-1. 
+1. **Removed Dash Prefix**: Updated `$ref` reference for `EmbeddingResponseData` schema.
+   - **Before**: `- $ref: "#/components/schemas/EmbeddingResponseData"`
+   - **After**: `$ref: "#/components/schemas/EmbeddingResponseData"`
```
Comment (ayeshLK) [inline docs/spec/sanitations.md]: Can you add a proper description on why we need to do this as well ?

File: docs/spec/sanitations.md [hunk 3]
```diff
@@ -1,16 +1,18 @@
...
 _Edition_: Swan Lake
 
 # Sanitation for OpenAPI specification
 
 This document records the sanitation done on top of the official OpenAPI specification from Mistral AI. 
-The OpenAPI specification is obtained from (TODO: Add source link).
+The OpenAPI specification is obtained from [Mistral AI OpenAPI specification](https://docs.mistral.ai/redocusaurus/plugin-redoc-0.yaml).
 These changes are done in order to improve the overall usability, and as workarounds for some known language limitations.
 
 [//]: # (TODO: Add sanitation details)
```
Comment (ayeshLK) [inline docs/spec/sanitations.md]: Lets remove this.

File: docs/spec/sanitations.md [hunk 4]
```diff
@@ -1,16 +1,18 @@
...
-1. 
+1. **Removed Dash Prefix**: Updated `$ref` reference for `EmbeddingResponseData` schema.
+   - **Before**: `- $ref: "#/components/schemas/EmbeddingResponseData"`
+   - **After**: `$ref: "#/components/schemas/EmbeddingResponseData"` 
 2. 
 3.
```
Comment (ayeshLK) [inline docs/spec/sanitations.md]: If we do not have any more sanitations lets remove these.

File: ballerina/tests/mock_service.bal [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+function init() returns error? {
+    if isLiveServer {
+        log:printInfo("Skiping mock server initialization as the tests are running on live server");
+        return;
+    }
+    log:printInfo("Initiating mock server");
+    check httpListener.attach(mockService, "/");
+    check httpListener.'start();
+}
```
Comment (ayeshLK) [inline ballerina/tests/mock_service.bal]:
```suggestion
}

```

File: ballerina/tests/README.md
Comment (ayeshLK) [inline ballerina/tests/README.md:1]: This file is not needed as the build, test, and release is running on gradle.

**Review by ayeshLK — CHANGES_REQUESTED** [review]

File: ballerina/tests/README.md
Comment (thiunuwan) [reply ballerina/tests/README.md:1]: Resolved in commit https://github.com/ballerina-platform/module-ballerinax-mistral/pull/3/commits/f4fe9e1b35b6a9c171ab9beeb25ee76fff78614c

File: ballerina/tests/mock_service.bal (hunk 5 above)
Comment (thiunuwan) [reply ballerina/tests/mock_service.bal]: Resolved in commit https://github.com/ballerina-platform/module-ballerinax-mistral/pull/3/commits/f4fe9e1b35b6a9c171ab9beeb25ee76fff78614c

File: docs/spec/sanitations.md (hunk 4 above)
Comment (thiunuwan) [reply docs/spec/sanitations.md]: Resolved in commit https://github.com/ballerina-platform/module-ballerinax-mistral/pull/3/commits/f4fe9e1b35b6a9c171ab9beeb25ee76fff78614c

File: docs/spec/sanitations.md (hunk 3 above)
Comment (thiunuwan) [reply docs/spec/sanitations.md]: Resolved in commit https://github.com/ballerina-platform/module-ballerinax-mistral/pull/3/commits/f4fe9e1b35b6a9c171ab9beeb25ee76fff78614c

File: docs/spec/sanitations.md (hunk 2 above)
Comment (thiunuwan) [reply docs/spec/sanitations.md]: Resolved in commit https://github.com/ballerina-platform/module-ballerinax-mistral/pull/3/commits/f4fe9e1b35b6a9c171ab9beeb25ee76fff78614c

**Comment by NipunaRanasinghe** [comment]
Closing due to inactivity. @thiunuwan Please feel free to re-open after addressing the suggestions.

**Pull request closed without merging**